src/main/java/parser/HttpRequestParser.java

```diff
@@ -0,0 +1,112 @@
+
+import http.HttpHeader;
+import http.HttpMethod;
+import http.HttpRequestBody;
+import http.HttpRequestLine;
+import http.URI;
+import utils.IOUtils;
+
+public class HttpRequestParser {
+	private static final String QUERY_SEPARATOR = "?";
+	private static final String SPACE = " ";
+	private static final String HEADER_SEPARATOR = ": ";
+	private static final String PARAMETER_EQUAL_SIGN = "=";
+	private static final String HEADER_CONTENT_LENGTH = "Content-Length";
+	private static final String HEADER_CONTENT_TYPE = "Content-Type";
+	private static final String CONTENT_TYPE_FORM = "application/x-www-form-urlencoded";
+
+	private static final int HTTP_METHOD_LOCATION = 0;
+	private static final int HTTP_PATH_LOCATION = 1;
+	private static final int HTTP_PROTOCOL_LOCATION = 2;
+	private static final int KEY_LOCATION = 0;
+	private static final int VALUE_LOCATION = 1;
+
+	private HttpBodyParsingStrategy httpBodyParsingStrategy;
+
+	private final BufferedReader bufferedReader;
+
+	public HttpRequestParser(BufferedReader bufferedReader) {
+		this.bufferedReader = bufferedReader;
+		this.httpBodyParsingStrategy = null;
+	}
+
+	private HttpRequestBody parseHttpBody(String bodyString) {
+		return httpBodyParsingStrategy.parse(bodyString);
+	}
+
+	public HttpRequestLine parseHttpRequestLine() throws IOException {
+		String[] requestLine = this.bufferedReader.readLine().split(SPACE);
+		HttpMethod httpMethod = HttpMethod.valueOf(requestLine[HTTP_METHOD_LOCATION]);
+		String path = requestLine[HTTP_PATH_LOCATION];
+		String protocol = requestLine[HTTP_PROTOCOL_LOCATION];
+
+		return new HttpRequestLine(makeURIFromPath(path), httpMethod, protocol);
+	}
+
+	public HttpHeader parseHttpHeader() throws IOException {
+		HttpHeader httpHeader = new HttpHeader();
+
+		String line;
+		while ((line = this.bufferedReader.readLine()) != null && !line.isEmpty()) {
+			String[] header = line.split(HEADER_SEPARATOR);
+			httpHeader.addHeader(header[KEY_LOCATION], header[VALUE_LOCATION]);
+		}
+
+		return httpHeader;
+	}
+
+	public HttpRequestBody parseHttpRequestBody(HttpHeader httpHeader) throws IOException {
+		if (!(httpHeader.containsKey(HEADER_CONTENT_TYPE) && httpHeader.containsKey(HEADER_CONTENT_LENGTH))) {
+			return new HttpRequestBody();
+		}
+		if (CONTENT_TYPE_FORM.equals(httpHeader.getValueByKey(HEADER_CONTENT_TYPE))) {
+			this.httpBodyParsingStrategy = new FormUrlEncodedParsingStrategy();
+			int contentLength = Integer.parseInt(httpHeader.getValueByKey(HEADER_CONTENT_LENGTH));
+			return parseHttpBody(IOUtils.readData(bufferedReader, contentLength));
+		}
+		throw new IllegalArgumentException("지원되지 않는 컨텐츠타입입니다.");
+	}
+
+	private URI makeURIFromPath(String path) {
+		Map<String, String> parameters = new HashMap<>();
+		if (hasQuery(path)) {
+			int queryStartIndex = path.indexOf(QUERY_SEPARATOR);
+			path = path.substring(0, queryStartIndex);
+			parameters = parseParameters(path);
+		}
+		return new URI(path, parameters);
+	}
```

Comment:
가급적 변수를 할당함과 동시에 선언하도록해보시면 좋을 것 같아요. 🙂
```suggestion
	private URI makeURIFromPath(String path) {
		if (hasQuery(path)) {
			int queryStartIndex = path.indexOf(QUERY_SEPARATOR);
			path = path.substring(0, queryStartIndex);
			Map<String, String> parameters = parseParameters(path);

			return new URI(path, parameters);
		}
		
		return new URI(path, Collections.emptyMap());
	}
```